cni-plugin/pkg/plugin/plugin.go

```diff
@@ -260,7 +261,21 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 	}
 
 	// Check if there's an existing endpoint by listing the existing endpoints based on the WEP name prefix.
-	endpoints, err := calicoClient.WorkloadEndpoints().List(ctx, options.ListOptions{Name: wepPrefix, Namespace: wepIDs.Namespace, Prefix: true})
+
+	// We know that, in KDD, even though there may be >1 endpoint, we're only
+	// looking up one backing Pod.  Send it a hint that we really want it to
+	// do a Get instead of a list.  CNI plugin only has RBAC permissions to do
+	// a get on Pod resources.  The default used to be to do a Get if possible,
```

Comment:
>  CNI plugin only has RBAC permissions

Any reason to not just switch to a List() with the field selector specified, and give the CNI plugin the necessary permissions?

Reply:
Mainly just thinking that it was quite a big loosening to give CNI plugin list permissions when it could make do with get before.  That, and the fact that I'd need to adjust manifests and operator to loosen permissions, which felt like extra work to go in the wrong direction.

Reply:
I suppose I just don't see a huge security distinction between "Get any pod in the cluster" vs "List any pod in the cluster" 

But I'm happy to leave it as is.